packages/syft/src/syft/service/request/request_service.py

```diff
@@ -179,55 +181,64 @@ def apply(
         )
         request_notification = filter_by_obj(context=context, obj_uid=uid).unwrap()
 
-        link = LinkedObject.with_context(request, context=context)
-
         if not request.get_status(context) == RequestStatus.PENDING:
             if request_notification is not None:
                 mark_as_read = context.server.get_service_method(
                     NotificationService.mark_as_read
                 )
                 mark_as_read(context=context, uid=request_notification.id)
 
-                notification = CreateNotification(
-                    subject=f"Your request ({str(uid)[:4]}) has been approved!",
-                    from_user_verify_key=context.credentials,
+                self._send_email_notification(
+                    context=context,
+                    message_subject=f"Your request ({str(uid)[:4]}) has been approved. ",
+                    request=request,
                     to_user_verify_key=request.requesting_user_verify_key,
-                    linked_obj=link,
-                    notifier_types=[NOTIFIERS.EMAIL],
                     email_template=RequestUpdateEmailTemplate,
                 )
+        return result
 
-                send_notification = context.server.get_service_method(
-                    NotificationService.send
-                )
-                send_notification(context=context, notification=notification)
+    def _send_email_notification(
```

Comment:
wouldnt it be better to resultify this and do the error handling from where this is called? By implementing it like this we remove the ability to build logic on it in the case of an error